### PR TITLE
Deploy with ArgoCD

### PR DESCRIPTION
**Pull request by garryod**
Switched from one monolithic chart (`charts/developer-portal`) to an [app-of-apps](https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/#app-of-apps-pattern) (`charts/apps`), frontend deployment (`charts/frontend`) and backend deployment (`charts/backend`) with postgres deployed using the [bitnami chart](https://github.com/bitnami/charts/tree/main/bitnami/postgresql).

### CODE REVIEW

File: charts/frontend/templates/deployment.yaml
Comment (callumforrester) [inline charts/frontend/templates/deployment.yaml:1]: I don't know how useful this is but is there any point in making this a git move from `charts/developer-portal/templates/frontend-deployment.yaml` to preserve history? Same for the other templates.

File: README.md [hunk 1]
```diff
@@ -27,14 +27,12 @@ The Diamond Light Source developer portal, built with backstage for deployment o
...
-  - Populating the [necessary secrets](#necessary-secrets) in their respective fields in `backend.auth`
-  - Pointing to an existing secret containing the [necessary secrets](#necessary-secrets) with `backend.auth.existingSecret` (recommended)
+- Adjust configuration in `app.yaml` and `charts/apps/values.yaml` as appropriate
+- If deploying to non `dev-portal` namespace, create a secret containing the [necessary secrets](#necessary-secrets) with `backend.auth.existingSecret` (recommended)
 - Deploy the portal, with:
 
   ```sh
-  cd charts/developer-portal
-  helm install <deployment-name> .
+  kubectl apply -f app.yaml
```
Comment (callumforrester) [inline README.md:35]: Shouldn't this be a `helm install` command for the app of apps?

**Review by callumforrester — APPROVED** [review]
LGTM, but I'm no expert!

File: app.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+metadata:
+  name: developer-portal
+  namespace: dev-portal
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+spec:
+  project: dev-portal
+  source:
+    repoURL: https://github.com/diamondlightsource/developer-portal
+    targetRevision: deploy-with-argocd
```
[the remark this answers is not in the document]
Comment (garryod) [reply app.yaml:12]: Good catch, plan was to remove `6c266dbdf93d34d0a94ad59a1e4856e4ca89fd9a` (named "Remove me") before merging

File: charts/frontend/templates/deployment.yaml
Comment (garryod) [reply charts/frontend/templates/deployment.yaml:1]: Not really, it's a completely different chart

File: README.md (hunk 1 above)
Comment (garryod) [reply README.md:35]: Nah, you just gotta' apply the single app manifest and let ArgoCD deploy the app of apps chart

**Review by callumforrester — APPROVED** [review]

**Pull request merged**